language/compiler/ir_to_bytecode/src/compiler.rs

```diff
@@ -925,8 +925,21 @@ fn compile_expression(
             let type_actuals_id = context.locals_signature_index(tokens)?;
             let def_idx = context.struct_definition_index(&name)?;
 
+            let self_name = ModuleName::new(ModuleName::SELF.to_string());
+            let ident = QualifiedStructIdent {
+                module: self_name,
+                name,
+            };
+            let sh_idx = context.struct_handle_index(ident)?;
+
             let num_fields = fields.len();
-            for (_, e) in fields {
+            for (field_order, (field, e)) in fields.into_iter().enumerate() {
+                // Check that the fields are specified in order matching the definition.
+                let (_, _, decl_order) = context.field(sh_idx, field.clone())?;
+                if field_order != decl_order {
+                    bail!("field `{}' defined out of order", field);
```

Comment:
Nit, maybe include the name of the struct in the error message 
i.e.
`"Field {} defined out of order for struct {}", field, name`

The main reason being is we don't have any line information currently

Reply:
Not blocking for this, don't really care about this. 

Just for the test :)

Reply:
OK.